SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.h

```diff
@@ -0,0 +1,207 @@
+  OUT       UINT8                  *Value
+  );
+
+/**
+  Get the full setting in binary from FV.
+
+  @param This       Seting Provider
+  @param ValueSize  IN=Size of location to store value
+                    OUT=Size of value stored
+  @param Value      Output parameter for the setting value.
+                    The type and size is based on the provider type
+                    and must be allocated by the caller.
+  @retval EFI_SUCCESS           If setting could be retrieved.
+  @retval EFI_BUFFER_TOO_SMALL  If the ValueSize on input is too small
+  @retval ERROR                 Error
+**/
+EFI_STATUS
+EFIAPI
+ConfDataGet (
+  IN CONST DFCI_SETTING_PROVIDER  *This,
+  IN OUT   UINTN                  *ValueSize,
+  OUT      UINT8                  *Value
+  );
+
+/**
+  Set configuration to default value from UEFI FV.
+
+  @param This          Seting Provider protocol
+
+  @retval EFI_SUCCESS  default set
+  @retval ERROR        Error
+**/
+EFI_STATUS
+EFIAPI
+ConfDataSetDefault (
+  IN  CONST DFCI_SETTING_PROVIDER  *This
+  );
+
+/**
+  Set new configuration value to variable storage.
+
+  @param This      Provider Setting
+  @param Value     a pointer to a datatype defined by the Type for this setting.
+  @param ValueSize Size of the data for this setting.
+  @param Flags     Informational Flags passed to the SET and/or Returned as a result of the set
+
+  @retval EFI_SUCCESS If setting could be set.  Check flags for other info (reset required, etc)
+  @retval Error       Setting not set.
+**/
+EFI_STATUS
+EFIAPI
+ConfDataSet (
+  IN  CONST DFCI_SETTING_PROVIDER  *This,
+  IN        UINTN                  ValueSize,
+  IN  CONST UINT8                  *Value,
+  OUT DFCI_SETTING_FLAGS           *Flags
+  );
+
+/**
+  Get the default value of a single setting from UEFI FV.
+  This getter will serialize default configuration setting
+  to printable strings to be used in Config App.
+
+  @param This           Seting Provider
+  @param ValueSize      IN=Size of location to store value
+                        OUT=Size of value stored
+  @param DefaultValue   Output parameter for the settings default value.
+                        The type and size is based on the provider type
+                        and must be allocated by the caller.
+
+  @retval EFI_SUCCESS           If the default could be returned.
+  @retval EFI_BUFFER_TOO_SMALL  If the ValueSize on input is too small
+  @retval ERROR                 Error
+**/
+EFI_STATUS
+EFIAPI
+SingleConfDataGetDefault (
+  IN  CONST DFCI_SETTING_PROVIDER  *This,
+  IN  OUT   UINTN                  *ValueSize,
+  OUT       UINT8                  *Value
+  );
+
+/**
+  Set new configuration value to variable storage.
+
+  @param This      Provider Setting
+  @param Value     a pointer to a datatype defined by the Type for this setting.
+  @param ValueSize Size of the data for this setting.
+  @param Flags     Informational Flags passed to the SET and/or Returned as a result of the set

```

Comment:
This parameter isn't included in the actual function.

Reply:
Updated.